### DISCUSSION TITLE
Create HTML/CSS/JS chunk literal?

### DISCUSSION THREAD

**Discussion by MayaGans** [quarto-dev/quarto-cli#1871] [Q&A] [answered]
Hi! Sorry if this is already documented somewhere but I'm writing up some blog posts where I want to display HTML, CSS, and JS code but not evaluate it! In .Rmd I used to cheat and use ` ``` ` without any code language (or throw in js, css, or html just to get the code highlighting I want) but I'm not sure what I should do in quarto? [I think quarto hates a chunk without any language declared] I can try hacking it and adding my own code highlighter but first wanted to check in and see if this functionality exists already 

Example: (pretty much want this to look exactly how it does in github rendering

```
<div>test</div>
```

**Comment by jjallaire** [comment, marked as answer]
In Quarto you should be able to do this:

````
```
<div>test</div>
```
````

Or this:

````
```html
<div>test</div>
```
````

Or this to get the grey box without highlighting:

````
```default
<div>test</div>
```
````

**Reply by MayaGans** [reply]
Ah ha! No curlies around `html`! Thank you!!!!!

**Reply by cscheid** [reply]
I don't know if we have this documented anywhere, but the conventions, afaik are the following:

## Code cells: ` ``` `

Single curly braces imply executable content

````
```{python}
This will run
```
````

No curly braces imply syntax highlighting

````
```python
print("Hello, world") # this will be highlighted as python code
```
````

_double_ curly braces imply "fenced output", which comes up when you're teaching quarto, and we will emit syntax-highlighted code while keeping the triple ticks and (single) curly brackets:

````
```{{python}}
print("Hello, world") # this will also be highlighted as python code
```
````

## Pandoc divs: `:::`

No curly braces imply a _class_:

```
::: some-class
:::
```

Curly braces imply a mixture of any number of classes and attributes, but then classes have a slightly different syntax:

```
::: {.some-class attribute="value"}
:::
```

These classes have a variety of uses:

* Filters can detect classes and treat that output differently. This is used throughout quarto (eg for conditional content, layout, etc)
  * (The set of classes that are detected by quarto should be better documented on the webpage, we're working on it!)
* If filters don't process classes or attributes, then they're passed on to the format. In HTML output, they're exactly CSS classes and DOM attributes.
